pydm/data_plugins/__init__.py

```diff
@@ -129,42 +270,14 @@ def load_plugins_from_path(locations, token):
 
     Returns
     -------
-    plugins: dict
+    plugins : dict
         Dictionary of plugins add from this folder
     """
     added_plugins = dict()
     for loc in locations:
-        for root, _, files in os.walk(loc):
-            if root.split(os.path.sep)[-1].startswith("__"):
-                continue
-
-            logger.debug("Looking for PyDM Data Plugins at: %s", root)
-            for name in files:
-                if name.endswith(token):
-                    try:
-                        logger.debug("Trying to load %s...", name)
-                        sys.path.append(root)
-                        temp_name = str(uuid.uuid4())
-                        module = imp.load_source(temp_name,
-                                                 os.path.join(root, name))
-                    except Exception as e:
-                        logger.exception("Unable to import plugin file %s."
-                                         "This plugin will be skipped."
-                                         "The exception raised was: %s",
-                                         name, e)
-                        continue
-                    classes = [obj for name, obj in inspect.getmembers(module)
-                               if (inspect.isclass(obj)
-                                   and issubclass(obj, PyDMPlugin)
-                                   and obj is not PyDMPlugin)]
-                    # De-duplicate classes.
-                    classes = list(set(classes))
-                    for plugin in classes:
-                        if plugin.protocol is not None:
-                            # Add to global plugin list
-                            add_plugin(plugin)
-                            # Add to return dictionary of added plugins
-                            added_plugins[plugin.protocol] = plugin
+        for plugin in find_plugins_from_path(loc, token=token):
+            if plugin.protocol is not None:
+                added_plugins[plugin.protocol] = add_plugin(plugin)
```

Comment:
Same as above: good to add a warning here in the else block

Reply:
Resolved in 52849a12ac00a0a4d6afafb2141ec7581d8ea5ae